library/middleware/shouldBlockRequest.ts

```diff
@@ -21,6 +21,11 @@ export function shouldBlockRequest(): Result {
     return { block: false };
   }
 
+  if (context.executedMiddleware) {
+    logWarningAlreadyExecutedMiddleware();
+    return { block: false };
+  }
```

Comment:
Should be fine?

Reply:
Should we add an early return? Would be nice if we could return early and remove the logged warning, but what if the user adds the middleware twice and the `Zen.setUser` call is in between of both? In this case we would never do user blocking.